### PR TITLE
Update to latest version of Web Bundles spec

### PR DESCRIPTION
**Pull request by felipeerias**
Remove primary URL and manifest from the WBN library, following the Web Bundles spec.

https://wpack-wg.github.io/bundled-responses/draft-ietf-wpack-bundled-responses.html

Issue #686

### CODE REVIEW

**Comment by felipeerias** [comment]
This PR removes support for primary URL and manifest from the WBN library, following [the most recent version of the Web Bundles spec](https://wpack-wg.github.io/bundled-responses/draft-ietf-wpack-bundled-responses.html).

Should the version field be updated as well?

In my system, all the test passed except the one that reads `testdata/hello.wbn`. I guess that this file should be regenerated with the updated encoder?

**Comment by hayatoito** [comment]
Great! Thanks for working on this!
@irori Could you please review this PR?

**Comment by irori** [comment]
Thanks for the PR!

It would be nice if we could keep support for the old format (as `gen-bundle` does). Are you interested in doing that in this PR? If not, I will create a follow-up PR.

> Should the version field be updated as well?

Yes, `gen-bundle` uses `b2\0\0` for the new version of the format. `wbn` should align with that.

> In my system, all the test passed except the one that reads testdata/hello.wbn. I guess that this file should be regenerated with the updated encoder?

Right, `testdata/hello.wbn` should be regenerated with the latest `gen-bundle`, specifying `-version b2` command-line option.

File: js/bundle/src/decoder.ts [hunk 1]
```diff
@@ -80,28 +94,86 @@ export class Bundle {
...
+        constructor(private bundle: Bundle) {
+        }
+
+        getResponse(url: string): Response {
+          const indexEntry = asArray(this.bundle.indexSection[url]);
+          if (!indexEntry) {
+            throw new Error('No entry for ' + url);
+          }
+          const [offset, length] = indexEntry;
+          if (indexEntry.length !== 2) { // this.compatAdapter.responseEntryLength
```
Comment (irori) [inline js/bundle/src/decoder.ts:161]: Forgot to remove this comment?

File: js/bundle/src/cli.ts [hunk 2]
```diff
@@ -6,19 +6,32 @@ const options = commander
...
+
+if (options.formatVersion === undefined || options.formatVersion === 'b2') {
+  // webbundle format version b2
+  const builder = new BundleBuilder();
+  builder.addFilesRecursively(options.baseURL, options.dir);
+  fs.writeFileSync(options.output, builder.createBundle());
+} else if (options.formatVersion === 'b1') {
+  // webbundle format version b1
+  const primaryURL = options.primaryURL || options.baseURL;
+  const builder = new BundleBuilder(primaryURL);
```
[the remark this answers is not in the document]
Comment (felipeerias) [reply js/bundle/src/cli.ts:28]: Yes, fixed.

**Pull request merged**